File: app/src/main/java/x/chestnut/code/snippet/ui/recyclerView/xItemPackage/XItemFragment.java
package x.chestnut.code.snippet.ui.recyclerView.xItemPackage;

import android.view.View;

import x.chestnut.code.snippet.base.ScrollBaseFragment;
import x.chestnut.code.snippet.ui.recyclerView.xItemPackage.baseUse.XItemBaseUseFragment;
import x.chestnut.code.snippet.ui.recyclerView.xItemPackage.centerScale.CenterScaleFragment;
import x.chestnut.code.snippet.ui.recyclerView.xItemPackage.circleScrollRecyclerView.CircleRecyclerViewFragment;

/**
 * <pre>
 *     author: Chestnut
 *     blog  : https://juejin.im/user/676954892408824
 *     time  : 2019/4/2 23:56
 *     desc  :
 *     thanks To:
 *     dependent on:
 *     update log:
 * </pre>
 */

public class XItemFragment extends ScrollBaseFragment{

    public XItemFragment() {

    }

    public static XItemFragment newInstance() {
        return new XItemFragment();
    }

    @Override
    protected void onLazyViewCreate(View rootView) {
        addView("基本用法-轻松实现多Item布局",view ->
                startFragment(XItemBaseUseFragment.newInstance()));
        addView("无限循环RecyclerView",view ->
                startFragment(CircleRecyclerViewFragment.newInstance()));
        addView("中心放大RecyclerView",view ->
                startFragment(CenterScaleFragment.newInstance()));
    }

    @Override
    protected void onViewResume() {
        setTitle("X-Item封装");
    }
}

File: app/src/main/java/x/chestnut/code/snippet/ui/recyclerView/xItemPackage/XItemFragment.kt
package x.chestnut.code.snippet.ui.recyclerView.xItemPackage

import android.view.View
import x.chestnut.code.snippet.base.ScrollBaseFragment
import x.chestnut.code.snippet.ui.recyclerView.xItemPackage.baseUse.XItemBaseUseFragment
import x.chestnut.code.snippet.ui.recyclerView.xItemPackage.centerScale.CenterScaleFragment
import x.chestnut.code.snippet.ui.recyclerView.xItemPackage.circleScrollRecyclerView.CircleRecyclerViewFragment

/**
 * <pre>
 * author: Chestnut
 * blog  : https://juejin.im/user/676954892408824
 * time  : 2019/4/2 23:56
 * desc  :
 * thanks To:
 * dependent on:
 * update log:
</pre> *
 */
class XItemFragment : ScrollBaseFragment() {

    override fun onLazyViewCreate(rootView: View) {
        addView("基本用法-轻松实现多Item布局") { startFragment(XItemBaseUseFragment()) }
        addView("无限循环RecyclerView") { startFragment(CircleRecyclerViewFragment()) }
        addView("中心放大RecyclerView") { startFragment(CenterScaleFragment()) }
    }

    override val actionBarTitle: String
        get() = "X-Item封装"
}